src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -117,34 +117,36 @@ private TestSuite Build(Assembly assembly, string assemblyNameOrPath, IDictionar
                 {
                     // This cannot be changed without breaking backwards compatibility with old runners.
                     // Deserializes the way old runners understand.
-                    
-                    options.TryGetValue(FrameworkPackageSettings.TestParameters, out object testParameters);
-                    string parametersString = testParameters as string;
-                    if (!string.IsNullOrEmpty(parametersString))
+
+                    if (options.TryGetValue(FrameworkPackageSettings.TestParameters, out var testParameters))
                     {
-                        foreach (string param in parametersString.Split(new[] { ';' }))
+                        var parametersString = (string?)testParameters;
+                        if (!string.IsNullOrEmpty(parametersString))
                         {
-                            int eq = param.IndexOf('=');
-                        
-                            if (eq > 0 && eq < param.Length - 1)
+                            foreach (var param in parametersString!.Split(';'))
```

Comment:
I'm unsure, but I don't think this will route to the `Split(char, StringSplitOptions = None)` overload one might expect, since that's only on NETStandard2.1+. I think it will still route to the params overload. Though I like how this allows the codebase to automatically target the newer API if a netstandard21 or higher version of the framework were ever released